### PR TITLE
CI run black formatter on notebooks as well as Python scripts

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Hi :wave:

I've noticed that in your `.pre-commit-config` file, you check that your Python files follow the `black` formatting style, but you don't do so for your Jupyter Notebooks.

So, here's a little PR to add a hook for that!

### CODE REVIEW

**Comment by mloning** [comment]
Thanks @MarcoGorelli, could we also add the hooks for isort and flake8?

**Comment by MarcoGorelli** [comment]
Hi @mloning - sure, I'll give that a go tomorrow!

**Comment by MarcoGorelli** [comment]
I just had a look at running `nbqa-flake8` - there's a few errors because of
```
from sktime.forecasting.all import *
```

Would you like me to silence the `F405` when running flake8 on notebooks, or should the `01_forecasting` notebook's imports be re-written to be explicit? I'd suggest adding it in a separate PR so there's not too many changes to review in one go (likewise for adding `nbqa-isort`), but I can add it here if you prefer

**Comment by MarcoGorelli** [comment]
@mloning I've added `nbqa-isort` as well, as the diff's quite small.

For `nbqa-flake8`, I was trying to re-run the `01_forecasting` notebook (so I could replace the star import), but when trying to install `sktime` locally I get
```
(sktime-dev) marco@marco-Predator-PH315-52:~/sktime-dev$ pip install --editable .
Obtaining file:///home/marco/sktime-dev
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /home/marco/miniconda/envs/sktime-dev/bin/python /home/marco/miniconda/envs/sktime-dev/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp55q5_ruq
         cwd: /home/marco/sktime-dev
    Complete output (32 lines):
    Traceback (most recent call last):
      File "setup.py", line 226, in check_package_status
        module = importlib.import_module(package)
      File "/home/marco/miniconda/envs/sktime-dev/lib/python3.8/importlib/__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
      File "<frozen importlib._bootstrap>", line 991, in _find_and_load
      File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
    ModuleNotFoundError: No module named 'numpy'
    Traceback (most recent call last):
      File "/home/marco/miniconda/envs/sktime-dev/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
        main()
      File "/home/marco/miniconda/envs/sktime-dev/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/home/marco/miniconda/envs/sktime-dev/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 133, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/tmp/pip-build-env-9zr87gf4/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 161, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/tmp/pip-build-env-9zr87gf4/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 253, in run_setup
        super(_BuildMetaLegacyBackend,
      File "/tmp/pip-build-env-9zr87gf4/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 145, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 319, in <module>
        setup_package()
      File "setup.py", line 309, in setup_package
        check_package_status(package, version)
      File "setup.py", line 255, in check_package_status
        raise ImportError(
    ImportError: numpy is not installed.
    sktime requires numpy >= 1.18.0.
    Installation instructions are available on the sktime website: https://www.sktime.org/en/latest/installation.html
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/marco/miniconda/envs/sktime-dev/bin/python /home/marco/miniconda/envs/sktime-dev/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp55q5_ruq Check the logs for full command output.
```
However, I do have `numpy` installed:
```
(sktime-dev) marco@marco-Predator-PH315-52:~/sktime-dev$ python
Python 3.8.5 (default, Sep  4 2020, 07:30:14) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.__version__
'1.19.2'
```

**Comment by MarcoGorelli** [comment]
Have added an `nbqa-flake8` check as well :tada:

I added `E402 (Module level import not at top of file)` to extend-ignore for this one because sometimes there are imports in the middle of notebooks and I'd imagine you want to keep it that way? That's what the `--treat-comment-as-code` flag to `isort` achieves

**Comment by mloning** [comment]
Thanks @MarcoGorelli, much appreciated! 

Not sure about the installation error, haven't seen that one before. Our `setup.py` follows scikit-learn's and all the code does is call `importlib.import_module("numpy")`, so not sure why that fails.

**Comment by mloning** [comment]
FYI We just merged another big PR which resulted in a number of merge conflicts, I wanted to take a look at the notebooks anyway, so I can resolve them at the same time.

**Comment by mloning** [comment]
Thanks @MarcoGorelli for fixing this, realised I added a few more conflicts with other PRs, git isn't great for notebooks ...

Will take a look at this now and then hopefully merge it later today!

**Comment by MarcoGorelli** [comment]
🤣 no worries! I'll take a look at the conflicts and'll update

**Comment by mloning** [comment]
@MarcoGorelli just pushed the changes, that should be it hopefully!

**Comment by mloning** [comment]
FYI I reverted back to standard imports and remove the imports from `sktime.forecasting.all`, I still like the idea of it but perhaps it's better to be more explicit in tutorial notebooks about the file structure where functions/classes come from. 

I'll leave `sktime.forecasting.all` though, import everything you need via `from sktime.forecasting.all import *` seems way to convenient once you're familiar with the toolbox.

**Comment by mloning** [comment]
Interesting, the new CI checks we now give the same error that you reported earlier, that numpy is missing even though it has been installed, will have to look into this https://dev.azure.com/mloning/sktime/_build/results?buildId=1337&view=logs&s=6884a131-87da-5381-61f3-d7acc3b91d76&j=e395bb75-778f-525f-dfb1-52e1f09bc325

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -32,3 +32,11 @@ repos:
   hooks:
   - id: check-manifest
     stages: [manual]
+
+- repo: https://github.com/nbQA-dev/nbQA
+  rev: 0.3.5
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:37]:
```suggestion
  rev: 0.3.6
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+isort = "setup.cfg"
+black = "pyproject.toml"
+
+
+[tool.nbqa.mutate]
+isort = 1
+black = 1
+
+[tool.nbqa.addopts]
+isort = ["--treat-comment-as-code", "# %%", "-p", "sktime"]
```
Comment (MarcoGorelli) [inline pyproject.toml:12]:
```suggestion
[tool.nbqa.mutate]
isort = 1
black = 1

[tool.nbqa.addopts]
isort = ["-p", "sktime"]
```

**Comment by MarcoGorelli** [comment]
Hey @mloning 

Sorry for the delay, but I finally got to the root of the cause - it's that if you have a `pyproject.toml` file present, then `pip install -e .` works differently. If you look at sklearn, they have a `[build-system]` section in [their `pyproject.toml` file](https://github.com/scikit-learn/scikit-learn/blob/master/pyproject.toml).

So, rather than changing how sktime is packaged, I've just moved configuration for `nbqa` over to the pre-commit file - all seems to works as expected now :tada:

**Review by mloning — APPROVED** [review]
Hi @MarcoGorelli thank you so much for resolving this! I'll leave the PR open for a bit longer in case others want to comment, but looks all good to me now.

**Comment by mloning** [comment]
Thanks again @MarcoGorelli, will merge now!

**Pull request merged**